### DISCUSSION TITLE
Pass `chains` as props in <ConnectButton.Custom>

### DISCUSSION THREAD

**Discussion by 0xIvan** [rainbow-me/rainbowkit#1811] [Ideas]
gm,

Currently in <ConnectButton.Custom>, the connected chain is passed as prop. Is it possible to pass all other chains from the config as well? The idea is to create a custom chain modal with their respective icons. `chains` returned from wagmi hooks don't have the iconUrl.

Thanks!

**Comment by 0xmkh** [comment]
@0xIvan You can pass custom chains by specifying `iconUrl` like [this](https://www.rainbowkit.com/docs/custom-chains) and then pass it to `getDefaultConfig` or `createConfig` from wagmi.

Here is an example.

```
const config = getDefaultConfig({
  appName: "RainbowKit Demo",
  projectId,
  chains: [{ ...mainnet, iconUrl: "<your_icon_url>" }],
});
```

**Reply by 0xIvan** [reply]
Ah ok, I assumed it wouldn't be specified in `chains` returned from wagmi because the default `iconUrl` from rainbowkit is not specified. But this works thanks!
